app/cmd/start.go

```diff
@@ -108,10 +105,54 @@ func start(c *cli.Context) error {
 	}
 
 	if tlsConfig != nil {
-		logrus.Info("Creating grpc server with mtls auth")
+		logrus.Info("Creating gRPC server with mtls auth")
 	} else {
-		logrus.Info("Creating grpc server with no auth")
+		logrus.Info("Creating gRPC server with no auth")
+	}
+
+	shutdownCh := make(chan error)
+
+	// Start proxy server
+	proxyAddress, err := getProxyAddress(listen)
+	if err != nil {
+		return err
+	}
+
+	proxy, err := proxy.NewProxy(logsDir, shutdownCh)
```

Comment:
I think the new Proxy server will be only for engine instance manager only which will communicate with its downstream replicas, but here we just start it regardless of whether it's engine or replica instance manager.

I feel we should skip proxy for instance manager replica pod. WDYT?

Reply:
I think we can enhance this later. Added comment.

Reply:
Sure, please create a ticket for this later.

Reply:
https://github.com/longhorn/longhorn/issues/3968

Reply:
After re-checking this part. It seems we do not really need the proxying for IM-R. And the IM-E proxy server should be able to directly use the replica gRPC clients instead.
Since it's the IM-R proxy server rather than the longhorn manager responsible for talking with the replica processes (for backup-related tasks).